File: 2024/Typescript/day1a.ts
import * as fs from "fs";

const REAL = "./day1.txt";
// const TEST = "./test.txt";

let content: string;
try {
  content = fs.readFileSync(REAL, "utf-8").trim();
} catch (err) {
  throw new Error(`Falid to parse the file: ${(err as Error).message}`);
}

let left: number[] = [];
let right: number[] = [];

for (let line of content.split("\n")) {
  let [l, r] = line.trim().split(/\s+/);
  left.push(Number(l));
  right.push(Number(r));
}

left.sort((a, b) => a - b);
right.sort((a, b) => a - b);

let total: number = 0;
for (let i = 0; i < left.length; i++) {
  total += Math.abs(left[i] - right[i]);
}

console.log(total);

File: 2024/Cplusplus/day1/day1a.cpp
#include <algorithm>
#include <fstream>
#include <iostream>
#include <vector>

using namespace std;

#define REAL "day1.txt"
#define TEST "test.txt"

int main(void) {
  ifstream file(REAL);

  if (!file) {
    cerr << "Error in Text file\n";
    return 1;
  }

  vector<int> left;
  vector<int> right;

  int left_val;
  int right_val;
  while (file >> left_val >> right_val) {
    left.push_back(left_val);
    right.push_back(right_val);
  }

  sort(left.begin(), left.end());
  sort(right.begin(), right.end());

  int total = 0;
  for (auto i{0u}; i < left.size(); i++) {
    total += std::abs(left[i] - right[i]);
  }

  cout << total << endl;
  return 0;
}
